src/components/project/handlers/set-department-id.handler.ts

```diff
@@ -46,19 +46,18 @@ export class SetDepartmentId implements IEventHandler<SubscribedEvent> {
         //TODO: determine if schema update should be applied to allow for transaction locks.
         `
         MATCH ()-[:departmentId]-(departmentIdPropertyNode:Property)
-        WHERE departmentIdPropertyNode.value STARTS WITH $departmentIdPrefix
+        WHERE departmentIdPropertyNode.value STARTS WITH "${departmentIdPrefix}"
         WITH collect(distinct(toInteger(right(departmentIdPropertyNode.value, 4)))) as listOfDepartmentIds
-        UNWIND [n IN range(1, 9999) WHERE NOT n IN listOfDepartmentIds] as listOfUnusedDepartmentIds
+        UNWIND [n IN range(11, 9999) WHERE NOT n IN listOfDepartmentIds] as listOfUnusedDepartmentIds
         WITH toString(min(listOfUnusedDepartmentIds)) AS nextIdBase
-        WITH '${departmentIdPrefix}' + substring("0000", 1, 4 - size(nextIdBase)) + nextIdBase as nextId
-        MATCH (project:Project {id: $projectId})
+        WITH "${departmentIdPrefix}" + substring("0000", 1, 4 - size(nextIdBase)) + nextIdBase AS nextId
+        MATCH (project:Project {id: "${project.id}"})
         OPTIONAL MATCH (project)-[oldDepartmentIdRelationship:departmentId {active: true}]-(oldDepartmentIdPropertyNode:Property)
         SET oldDepartmentIdRelationship.active = false
         WITH coalesce(oldDepartmentIdPropertyNode.value, nextId) as departmentId, project
         CREATE (project)-[newDepartmentIdRelationship:departmentId { active: true, createdAt: datetime() }]->(:Property { createdAt: datetime(), value: departmentId })
         RETURN departmentId
-        `,
-        { departmentIdPrefix: departmentIdPrefix, projectId: project.id }
+        `
```

Comment:
This is just concatenating the parameters into the query which is not what we want to do. We want parameters to be bound for safety & performance.

Parameters are only bound when using [`raw`](https://jamesfer.me/cypher-query-builder/classes/query.html#raw) as a template tag. i.e.
```ts
raw`match ... ${param}`
```
instead of a function call
```ts
raw(`match ${param}`) // <- string concatenation
```

Reply:
What does binding them do? How is it different?

Reply:
So it's easier for the engine to parse?

Reply:
[Yes](https://neo4j.com/docs/cypher-manual/current/syntax/parameters/) but it also prevents
things like `"${foo}"` where foo is something like `"; match (n) detach delete n; "`. I know these variables are not directly user input, but it's just good practice. And better for the database as the link states.

Reply:
Ok. Thanks for the info. Makes sense. At first I thought it was a problem with the query builder variable, but it was just not a string. I fixed the function that grabs the funding account number for this to actually return a string instead of a number.

I think it's ready to go then.